xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletURLFactory.java

```diff
@@ -490,6 +499,20 @@ public URL createResourceURL(String filename, boolean forceSkinAction, XWikiCont
         }
         path.append("resources");
         addFileName(path, filename, false, context);
+        if (queryParameters != null && !queryParameters.isEmpty()) {
+            path.append("?");
+            Iterator<Map.Entry<String, String>> entryIterator = queryParameters.entrySet().iterator();
+            while (entryIterator.hasNext()) {
+                Map.Entry<String, String> entry = entryIterator.next();
+                path.append(entry.getKey());
+                path.append("=");
+                path.append(entry.getValue());
```

Comment:
I hope that the passed queryParameters are already encoded :)

Reply:
I should definitely be encoded here and not in the Map.